modules/service/src/main/scala/lucuma/odb/service/GuideService.scala

```diff
@@ -598,15 +603,16 @@ object GuideService {
           asterism     <- ResultT(getAsterism(pid, obsInfo.id))
           tracking      = ObjectTracking.fromAsterism(asterism)
           candPeriod    = neededPeriods.tail.fold(neededPeriods.head)((a, b) => a.span(b))
-          candidates   <- ResultT(
-                           getAllCandidates(obsInfo.id, candPeriod.start, candPeriod.end, tracking, genInfo.centralWavelength, obsInfo.constraints)
-                          )
+          candidates   <- ResultT:
+                           tracking match
+                              case None    => Result.success(Nil).pure[F]
+                              case Some(t) => getAllCandidates(obsInfo.id, candPeriod.start, candPeriod.end, t, genInfo.centralWavelength, obsInfo.constraints)
           positions     = getPositions(obsInfo.availabilityAngles, genInfo.offsets)
-          neededLists  <- ResultT.fromResult(
-                            neededPeriods.traverse(p =>
-                              buildAvailabilityList(p, obsInfo, genInfo, genInfo.centralWavelength, asterism, tracking, candidates, positions)
-                            )
-                          )
+          neededLists  <- ResultT.fromResult:
+                            neededPeriods.traverse: p =>
+                              tracking match
+                                case None => ???
```

Comment:
Should this be a `Success` result as you did in `candidates` ?

Reply:
Yes, good catch.